src/main/kotlin/racingcar/Application.kt

```diff
@@ -1,5 +1,30 @@
 package racingcar
 
+import racingcar.common.RandomPowerStrategy
+import racingcar.domain.Cars
+import racingcar.domain.TrialCount
+import racingcar.view.InputView
+import racingcar.view.OutputView
+
+fun <T> retry(action: () -> T): T {
+    while (true) {
+        try {
+            return action.invoke()
+        } catch (e: IllegalArgumentException) {
+            OutputView.printException(e)
+        }
```

Comment:
```suggestion
        runCatching { return action.invoke() }
            .onFailure { e ->
                if (e is IllegalArgumentException) {
                    OutputView.printException(e)
                }
            }
```
You can use this as an alternative. To be honest, I'm not sure which one is better, so I'd like to discuss it together.

Reply:
Maybe this one makes the code more readable when we have more complex logic for each code block.
For my code I think it's easier and more readable to just use `try-catch`.
But thank you for the information. I didn't know about it.

```kotlin
fun main() {
    doSomething()
        .onSuccess {
            TODO("do something when success")
        }
        .onFailure {
            TODO("do something when fail")
        }
}

fun doSomething(): Result<Unit> {
    return runCatching {
        TODO("some logic might be failed")
    }
}
```